_posts/2022-09-07-storage-write-removal-before-conditional-termination.md

```diff
@@ -0,0 +1,161 @@
+---
+layout: post
+published: true
+title: 'Optimizer Bug Causing the Removal of Storage Writes Before Calls to Functions Conditionally Terminating the Transaction'
+date: '2022-09-07'
+author: Solidity Team
+category: Security Alerts
+---
+On September 5, 2022, a bug in Solidity's Yul optimizer was found by differential fuzzing.
+
+The bug was introduced in version 0.8.13 and [Solidity version 0.8.17](https://github.com/ethereum/solidity/releases/tag/v0.8.17), released on September 07, 2022, provides a fix. The bug is significantly easier to trigger with optimized via-IR code generation, but can theoretically also occur in optimized legacy code generation.
+
+We assigned the bug a severity of "medium/high".
+
+
+## Which Contracts are Affected?
+
+A prerequisite for a contract to be affected is the use of ``return(...)`` or ``stop()`` in an inline assembly block (note that ``return(...)`` here refers to the builtin inline assembly function, not to the ``return`` statement of Solidity). Such an inline assembly call will not return from the current function, but instead result in an early successful (i.e. non-reverting) termination of the entire external EVM call.
+
+If your contract doesn't contain any inline assembly block with ``return(...)`` or ``stop()`` statements, you're unaffected.
+
+If it does and this early termination happens conditionally within a function, the optimizer could incorrectly remove storage writes before calls to the function.
+
+More concretely, the bug can occur if a contract contains the following pattern:
+
+1. A storage write.
+2. A call to a function that conditionally terminates using inline assembly as described above, but may also regularly return to the caller.
```

Comment:
I'll skip the "continues", though - it can also be a ``return;`` or in Yul a ``leave``, in which case it merely "continues" in the sense of continuing on the call site...

Reply:
I'd still count it as continuing :)